lightstep/resource_metric_dashboard.go

```diff
@@ -126,46 +126,37 @@ func getGroupSchema(chartSchemaType ChartSchemaType) map[string]*schema.Schema {
 				Schema: getChartSchema(chartSchemaType),
 			},
 		},
-	}
-}
-
-func getChartSchema(chartSchemaType ChartSchemaType) map[string]*schema.Schema {
-
-	var querySchema map[string]*schema.Schema
-	if chartSchemaType == UnifiedChartSchema {
-		querySchema = getUnifiedQuerySchemaMap()
-		querySchema["dependency_map_options"] = &schema.Schema{
+		"text_panel": {
 			Type:     schema.TypeList,
 			Optional: true,
-			MaxItems: 1,
+			Computed: true, // the panels can be mutated individually; chart mutations should not trigger group updates
 			Elem: &schema.Resource{
-				Schema: map[string]*schema.Schema{
-					"scope": {
-						Type:         schema.TypeString,
-						Optional:     true,
-						ValidateFunc: validation.StringInSlice([]string{"all", "upstream", "downstream", "immediate"}, false),
-					},
-					"map_type": {
-						Type:         schema.TypeString,
-						Optional:     true,
-						ValidateFunc: validation.StringInSlice([]string{"service", "operation"}, false),
-					},
-				},
+				Schema: getTextPanelSchema(),
 			},
-		}
-	} else {
-		querySchema = getMetricQuerySchemaMap()
+		},
 	}
+}
 
+func getTextPanelSchema() map[string]*schema.Schema {
+	return mergeSchemas(
+		getPanelSchema(),
+		map[string]*schema.Schema{
+			"text": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	)
+}
```

Comment:
Split out the common chart + text panel schema elements into `getPanelSchema` so `"text"` is the only unique field to text panels.